src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/SocketInput.cs

```diff
@@ -72,6 +74,8 @@ public void IncomingData(byte[] buffer, int offset, int count)
         {
             lock (_sync)
             {
+                CheckDisposed();
```

Comment:
If we see any failures I would expect to see it here. While `IncomingStart/Complete` shouldn't ever be called after the socket is closed, there's no telling what a ConnectionFilter might do.